src/main/java/com/puppycrawl/tools/checkstyle/api/FileText.java

```diff
@@ -170,14 +171,18 @@ public FileText(FileText fileText) {
      * @param file the name of the file
      * @param lines the lines of the text, without terminators
      * @throws NullPointerException if the lines array is null
+     * @throws IOException if problem occurs while getting canonical path to file
+     * @deprecated Use {@link #FileText(File, String)} instead
+     *     in order to preserve the original line terminators.
      */
-    public FileText(File file, List<String> lines) {
+    @Deprecated
+    public FileText(File file, List<String> lines) throws IOException {
```

Comment:
I do not expect any chnages in API.
Please explain your fix.
I expect changes only in JavadocPackageCheck.java.

Reply:
Error with somehow invalid path could arise in other checks as well. I decided to remove possibility of such problem. We already read file contents when constructing FileText object. So it seems to me that getting correct file path at the same place would be logical.

As for FileText constructor you mentioned. I marked it as Deprecated because JavaDoc states:
>This process does not restore the original line terminators and should therefore be avoided.

Also it is used only by deprecated FileContents constructor:
https://github.com/checkstyle/checkstyle/blob/b07838ab8dc825e9be86e624e734408d3753ff5c/src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java#L77-L81
which in turn is used only by tests so API changes there should not harm anything.